### PR TITLE
Read external fields from file in restarted simulation

### PR DESCRIPTION
**Pull request by RemiLehe**
When restarting a simulation, the grid external field from file were not read. 
As a result, a restarted simulation (e.g. with the electrostatic solver, were the grid external fields are applied at every iteration) would suddenly not feature these fields (without any error or warning message).

- [x] Cleanup code: maybe make separate function to avoid code duplication
- [x] Add a docstring to the new function
- [x] Documentation: explicitly state that the grid external fields are applied at every iteration, with the electrostatic solver
- [x] Add a restart automated test, based on https://github.com/ECP-WarpX/WarpX/tree/development/Examples/Tests/LoadExternalField

### CODE REVIEW

File: Source/WarpX.H [hunk 1]
```diff
@@ -984,6 +984,8 @@ public:
          char field,
          int lev, PatchType patch_type);
 
+    void LoadExternalFieldsFromFile (int const lev);
```
Comment (RemiLehe) [inline Source/WarpX.H]:
```suggestion
    void LoadExternalFieldsFromFile (int lev);
```

File: Source/Initialization/WarpXInitData.cpp [hunk 2]
```diff
@@ -728,6 +728,9 @@ void
 WarpX::PostRestart ()
 {
     mypc->PostRestart();
+    for (int lev = 1; lev <= maxLevel(); ++lev) {
```
Comment (RemiLehe) [inline Source/Initialization/WarpXInitData.cpp]:
```suggestion
    for (int lev = 0; lev <= maxLevel(); ++lev) {
```

File: Examples/Tests/LoadExternalField/inputs_rz [hunk 3]
```diff
@@ -55,14 +55,11 @@ proton.do_not_deposit = 1
...
 
-#electron.injection_style = "SingleParticle"
-#electron.single_particle_pos = 0.0  0.2  2.5
-#electron.single_particle_u = 0.0  0.0  0.0
-#electron.single_particle_weight = 1.0
-#electron.mass = 1.0
-#electron.charge = -q_e*1.0e-20
-
 # Diagnostics
-diagnostics.diags_names = diag1
```
Comment (RemiLehe) [inline Examples/Tests/LoadExternalField/inputs_rz:66]: This test is now a restart test ; as a consequence, the checkpoints need to be saved.

File: Source/Initialization/WarpXInitData.cpp [hunk 4]
```diff
@@ -728,6 +728,9 @@ void
 WarpX::PostRestart ()
 {
     mypc->PostRestart();
```
Comment (RemiLehe) [inline Source/Initialization/WarpXInitData.cpp:730]: This now loads external fields from file when restarting a simulation

File: Source/Initialization/WarpXInitData.cpp [hunk 5]
```diff
@@ -901,34 +904,7 @@ WarpX::InitLevelData (int lev, Real /*time*/)
        }
     }
```
Comment (RemiLehe) [inline Source/Initialization/WarpXInitData.cpp:906]: Cleaning up by putting the code below in a separate function.

**Review by ax3l — APPROVED** [review]
Thank you for this fix, looks good!

Agreed to add the docstring / todo's in your PR description, but will approve already.

**Review by ax3l — APPROVED** [review]
Docs look great, thx.

**Pull request merged**